spec/models/course_spec.rb

```diff
@@ -641,4 +641,80 @@
     subject { create(:course) }
     its(:recruitment_cycle_year) { should eq('2019') }
   end
+
+  describe '#enrichments' do
+    describe '#find_or_initialize_draft' do
+      let(:course) { create(:course, enrichments: enrichments) }
+
+      copyable_enrichment_attributes =
+        %w[
+          about_course
+          course_length
+          fee_details
+          fee_international
+          fee_uk_eu
+          financial_support
+          how_school_placements_work
+          interview_process
+          other_requirements
+          personal_qualities
+          qualifications
+          salary_details
+        ].freeze
+
+      subject { course.enrichments.find_or_initialize_draft }
+
+      context 'no enrichments' do
+        let(:enrichments) { [] }
+
+        it "sets all enrichment attributes to be nil" do
+          expect(subject.attributes.slice(*copyable_enrichment_attributes)).to eq copyable_enrichment_attributes.map { |key| [key, nil] }.to_h
+        end
+
+        its(:id) { should be_nil }
+        its(:last_published_timestamp_utc) { should be_nil }
+        its(:status) { should eq 'draft' }
+      end
+
+      context 'with a draft enrichment' do
+        let(:enrichment) { build(:course_enrichment, :initial_draft) }
+        let(:enrichments) { [enrichment] }
+
+        it "chas all the same enrichment attributes as the initial draft" do
+          expect(subject.attributes.slice(*copyable_enrichment_attributes)).to eq enrichment.attributes.slice(*copyable_enrichment_attributes)
+        end
+
+        its(:id) { should_not be_nil }
+        its(:last_published_timestamp_utc) { should eq enrichment.last_published_timestamp_utc }
+        its(:status) { should eq 'draft' }
+      end
+
+      context 'with a published enrichment' do
+        let(:enrichment) { build(:course_enrichment, :published) }
+        let(:enrichments) { [enrichment] }
+
+        it "chas all the same enrichment attributes as the published" do
+          expect(subject.attributes.slice(*copyable_enrichment_attributes)).to eq enrichment.attributes.slice(*copyable_enrichment_attributes)
```

Comment:
And if we have the above new `let`s we can then make this test much easier to read:

```
           expect(enrichment_attributes)).to eq published_enrichment_attributes 
```